pkg/config/template.go

```diff
@@ -97,30 +115,58 @@ func (c *TemplateCMManager) CreateCMTemplates() error {
 		templateName := GetTemplateName(filename)
 		cmName := GetTempCMName(templateName, filename, templateData)
 		cm := &v1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: cmName,
-				Labels: map[string]string{
-					createByRehearse:  "true",
-					rehearseLabelPull: strconv.Itoa(c.prNumber),
-				},
-			},
-			Data: map[string]string{filename: templateData},
+			ObjectMeta: metav1.ObjectMeta{Name: cmName},
+			Data:       map[string]string{filename: templateData},
 		}
 
 		c.logger.WithFields(logrus.Fields{"template-name": templateName, "cm-name": cmName}).Info("creating rehearsal configMap for template")
-		if _, err := c.cmclient.Create(cm); err != nil {
-			if kerrors.IsAlreadyExists(err) {
-				if _, err := c.cmclient.Update(cm); err != nil {
-					errors = append(errors, fmt.Errorf("could not update existing configmap: %v", err))
-				}
-			} else {
-				errors = append(errors, err)
-			}
+		if err := c.createCM(cm); err != nil {
+			errors = append(errors, err)
 		}
 	}
 	return kutilerrors.NewAggregate(errors)
 }
 
+func (c *TemplateCMManager) CreateClusterProfiles(dir string, profiles []ClusterProfile) error {
+	var errs []error
+	for _, p := range profiles {
+		cm, err := genClusterProfileCM(dir, p)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		c.logger.WithFields(logrus.Fields{"cluster-profile": cm.ObjectMeta.Name}).Info("creating rehearsal cluster profile ConfigMap")
+		if err := c.createCM(cm); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	return kutilerrors.NewAggregate(errs)
+}
+
+func genClusterProfileCM(dir string, profile ClusterProfile) (*v1.ConfigMap, error) {
+	ret := &v1.ConfigMap{}
+	ret.ObjectMeta = metav1.ObjectMeta{
+		Name: fmt.Sprintf("rehearse-cluster-profile-%s-%s", profile.Name, profile.TreeHash[:5]),
+	}
+	ret.Data = map[string]string{}
```

Comment:
is this really needed?

Reply:
Isn't it `nil` when `ret` is initialized with `v1.ConfigMap{}`?